ueberdosis/tiptap
# feature(core): pass through paste event to pasteHandler `getAttributes`

## Please describe your changes

This PR passes through the `ClipboardEvent` coming from the actual paste event down to the `getAttributes` function inside the pasteEvent handler for `MarkPasteRule` and `NodePasteRule`s.

This allows us to do more checks on the paste without breaking out of the Tiptap priority system.

This way I also fixed an issue with the Link extension being to agressive as it was ignoring all priorities on extensions when doing the pasteHandling.

This will fix #4347 as it was the issue bringing this up to me.

## How did you accomplish your changes

- Inside the PasteHandlerPlugin I added variables to store the paste/drop events in
- Those are passed down to the handler from where they are passed to the `getAttributes` function
- After that, the paste and drop event are set back to empty events

## How have you tested your changes

I tested my changes locally to see if the YouTube extension works fine again while the linking from the link extension still works on paste.

## How can we verify your changes

1. Checkout this PR
2. Add the Link extension to the YouTube demo
3. Try pasting a normal link (should work as usual)
4. Try pasting a youtube video (this now should embed the video)

## Remarks

- This should be backwards compatible and non breaking as the new parameter is optional to use.

## Checklist


## Related issues

- fixes #4347
